### PR TITLE
promote dns image 1.22.12

### PR DESCRIPTION
**Pull request by yosshi825**
1.22.12 has been released, but the image has not been updated yet
- https://github.com/kubernetes/dns/releases/tag/1.22.12

## ref 
https://github.com/kubernetes/k8s.io/pull/3993

### CODE REVIEW

File: k8s.gcr.io/images/k8s-staging-dns/images.yaml [hunk 1]
```diff
@@ -90,6 +93,7 @@
     "sha256:974b9caa21ae26fb9857f3f80a434ccf05022ced51a2d84ff34528788eaf5331": ["1.22.9"]
     "sha256:b980edf0948382803d4e12f144eafa069cb1ea2b9b9ad0f365642167e5d5f762": ["1.22.10"]
     "sha256:ca0f273d55df875d82a586f0d9eee4915bd88d92af05e21305e4d00da73e79a6": ["1.22.11"]
+    "sha256:XXX": ["1.22.12"]
```
Comment (dpasiukevich) [inline k8s.gcr.io/images/k8s-staging-dns/images.yaml:96]: could you fix that pls?

File: k8s.gcr.io/images/k8s-staging-dns/images.yaml [hunk 2]
```diff
@@ -442,6 +460,7 @@
     "sha256:bc60c7d8b48ec0d48cd844b6f6f3d84776a19d543ed0af2415335f3ecbc6487e": ["1.22.9"]
     "sha256:c3f4cff132268503232b5fee5a7af5007456965f78713c20cf33c9a7e23253f1": ["1.22.10"]
     "sha256:2113f50e50d78d70b2b6a938478c34668b694b0b16604758015ad537bd1029ba": ["1.22.11"]
+    "sha256:XXX": ["1.22.12"]
```
Comment (dpasiukevich) [inline k8s.gcr.io/images/k8s-staging-dns/images.yaml:463]: could you fix that pls?

**Review by dpasiukevich — COMMENTED** [review]
Thanks for the PR!

File: k8s.gcr.io/images/k8s-staging-dns/images.yaml (hunk 1 above)
Comment (yosshi825) [reply k8s.gcr.io/images/k8s-staging-dns/images.yaml]: sorry. fixed below
https://github.com/kubernetes/k8s.io/pull/4301/commits/5adea0139668b88f65da61ea72fedec461a7babb

File: k8s.gcr.io/images/k8s-staging-dns/images.yaml (hunk 2 above)
Comment (yosshi825) [reply k8s.gcr.io/images/k8s-staging-dns/images.yaml]: ditto
https://github.com/kubernetes/k8s.io/pull/4301/commits/d5c14231925976c6e42f438d556e156c2738f8b3

File: k8s.gcr.io/images/k8s-staging-dns/images.yaml [hunk 3]
```diff
@@ -472,6 +491,7 @@
     "sha256:81e7dfb9f713fc2eb08d26bba9c44030e31a78c1109a4e113cc80ae14c5e4fe7": ["1.22.9"]
     "sha256:7c80457a9dd8faac631d90902c2146e1ec540c93137264e0388a43dc5325b30f": ["1.22.10"]
     "sha256:aa2bee794a4a2843344c0e415c8e06d1b660de0e7766609a0c005ee472c8f4db": ["1.22.11"]
+    "sha256:a71a79e921601d9418333ac595d31b31b2ea9fed39b25a0916b500879c246ce7": ["1.22.12"]
```
Comment (dpasiukevich) [inline k8s.gcr.io/images/k8s-staging-dns/images.yaml:494]:
```suggestion
    "sha256:dfe2d344dfe147f45cd75187004ea583c54f43645ca63a0986168437d9211b84": ["1.22.12"]
```

**Review by dpasiukevich — CHANGES_REQUESTED** [review]
Could you update the hash for k8s-dns-node-cache-amd64 and let's merge.

**Pull request merged**